### DISCUSSION TITLE
Add proxy support for dns query

### DISCUSSION THREAD

**Discussion by sino1641** [AdguardTeam/AdGuardHome#3795] [Q&A]
According to the poor Internet connection 
which means some famous public DOHs are blocked by a great firewall,
I am here to ask if there is able to add proxy support for a dns query.
I have already searched the forum but find nothing.
Looking forward to reply and forgiving my poor English.

**Comment by ameshkov** [comment]
@ainar-g do we have a SOCKS proxy feature request?

**Reply by ainar-g** [reply]
@ameshkov, #391 briefly mentions SOCKS in the comments, but it is then closed and merged into #1228, which currently doesn't mention SOCKS.

**Reply by ameshkov** [reply]
It'd be better to have a separate issue about that then.

**Reply by ameshkov** [reply]
We'd first need to add proxy support to dnsproxy (there's already a feature request about that).

**Reply by ainar-g** [reply]
#3796

**Comment by sino1641** [comment]
Much thx for your support!
I mean something like using a proxy to send and receive dns requests from specific dns server which is blocked by ISP's firewall(as you may know)
and sending the local IP for CDN in the meanwhile if EDNS client subnet is supported by the server.
So are we taking about the same thing🤣?
Also thx for forgiving my poor English.
